src/braket/circuits/circuit.py

```diff
@@ -13,12 +13,14 @@
 
 from __future__ import annotations
 
-from typing import Callable, Dict, Iterable, List, Optional, Tuple, Type, TypeVar, Union
+from numbers import Number
```

Comment:
Why only use `Number` here and `float` elsewhere?

Reply:
We later cast to a float within an angle. This is used for a general check to see that the value passed in is any sort of number that can be made to a float later.